quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/DebugHostnameSettingsResource.java

```diff
@@ -0,0 +1,144 @@
+        for (String key : ConstantsDebugHostname.RELEVANT_OPTIONS) {
+            addOption(key);
+        }
+
+    }
+
+    @GET
+    @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX)
+    @Produces(MediaType.TEXT_HTML)
+    public String debug(final @PathParam("realmName") String realmName) throws IOException, FreeMarkerException {
+        FreeMarkerProvider freeMarkerProvider = keycloakSession.getProvider(FreeMarkerProvider.class);
+        RealmModel realmModel = keycloakSession.realms().getRealmByName(realmName);
+
+        URI frontendUri = keycloakSession.getContext().getUri(UrlType.FRONTEND).getBaseUri();
+        URI backendUri = keycloakSession.getContext().getUri(UrlType.BACKEND).getBaseUri();
+        URI adminUri = keycloakSession.getContext().getUri(UrlType.ADMIN).getBaseUri();
+
+        String frontendTestUrl = getTest(realmModel, frontendUri);
+        String backendTestUrl = getTest(realmModel, backendUri);
+        String adminTestUrl = getTest(realmModel, adminUri);
+
+        Map<String, Object> attributes = new HashMap<>();
+        attributes.put("frontendUrl", frontendUri.toString());
+        attributes.put("backendUrl", backendUri.toString());
+        attributes.put("adminUrl", adminUri.toString());
+
+        attributes.put("realm", realmModel.getName());
+        attributes.put("realmUrl", realmModel.getAttribute("frontendUrl"));
+
+        attributes.put("frontendTestUrl", frontendTestUrl);
+        attributes.put("backendTestUrl", backendTestUrl);
+        attributes.put("adminTestUrl", adminTestUrl);
+
+        attributes.put("serverMode", Environment.isDevMode() ? "dev [start-dev]" : "production [start]");
+
+        attributes.put("config", this.allConfigPropertiesMap);
+        attributes.put("headers", getHeaders());
+
+        return freeMarkerProvider.processTemplate(
+                attributes,
+                "debug-hostname-settings.ftl",
+                keycloakSession.theme().getTheme("base", Theme.Type.LOGIN)
+        );
+    }
+
+    private void addOption(String key) {
+        String rawValue = Configuration.getRawValue("kc." + key);
+        if (rawValue != null && !rawValue.isEmpty()) {
+            this.allConfigPropertiesMap.put(key, rawValue);
+        }
+    }
+
+
+    private Map<String, String> getHeaders() {
+        Map<String, String> headers = new TreeMap<>();
+        HttpHeaders requestHeaders = keycloakSession.getContext().getRequestHeaders();
+        for (String h : ConstantsDebugHostname.RELEVANT_HEADERS) {
+            addProxyHeader(h, headers, requestHeaders);
+        }
+        return headers;
+    }
+
+    private void addProxyHeader(String header, Map<String, String> proxyHeaders, HttpHeaders requestHeaders) {
+        String value = requestHeaders.getHeaderString(header);
+        if (value != null && !value.isEmpty()) {
+            proxyHeaders.put(header, value);
+        }
+    }
+
+    @GET
+    @Path("/test")
```

Comment:
I noticed the test URLs actually don't work. This should fix it:
```suggestion
    @Path("/{realmName}/" + DEFAULT_PATH_SUFFIX + "/test")
```

Could please also add a test to `HostnameDistTest` for this?

Reply:
Thanks for the suggestion. Accepted and added to the code. Done!